lib/mix/lib/mix/tasks/test.ex

```diff
@@ -229,9 +229,18 @@ defmodule Mix.Tasks.Test do
   end
 
   defp parse_files([], test_paths) do
+    warn_about_incorrect_extensions_in(test_paths)
+
     test_paths
   end
 
+  defp warn_about_incorrect_extensions_in(test_paths) do
+    for file <- Mix.Utils.extract_files(test_paths, "*_test.ex") do
+      IO.puts "Warning: #{file} has incorrect extension and won't be run. " <>
```

Comment:
Thanks, @lexmag! I really appreciate the pointers :D I'll definitely keep this in mind when working with mix tasks